mlflow/sparkml.py

```diff
@@ -0,0 +1,116 @@
+"""
+MLflow integration for SparkML models.
+
+SparkML models are saved and loaded using native SparkML persistence.
+The models can be exported as pyfunc for out-of Spark deployment or it can be loaded back as Spark
+Transformer in order to score it in Spark. The pyfunc flavor instantiates SparkContext internally
+and reads the input data as Spark DataFrame prior to scoring.
+
+Note: SparkML model can not be exported as spark_udf as those two formats are incompatible.
+"""
+
+from __future__ import absolute_import
+
+import os
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.environment import CondaEnvironment
+from mlflow.models import Model
+
+import pyspark
+from pyspark.ml.pipeline import PipelineModel
+from pyspark.ml.base import Transformer
+
+FLAVOR_NAME = "sparkml"
+
+
+def log_model(spark_model, artifact_path, env, jars=None):
```

Comment:
Why is env required here? Also document its type.

Reply:
goo catch, it should not be